src/main/java/com/github/manolo8/darkbot/BackpageManager.java

```diff
@@ -67,11 +70,37 @@ private int validTick() {
     }
 
     private int sidKeepAlive() throws Exception {
-        HttpURLConnection conn = (HttpURLConnection) new URL(instance + "indexInternal.es?action=" + getRandomAction())
+        return getConnection("indexInternal.es?action=" + getRandomAction()).getResponseCode();
+    }
+
+    public HttpURLConnection getConnection(String params) throws Exception{
+        if (isInvalid()) throw new UnsupportedOperationException("Can't connect when sid is invalid");
+        HttpURLConnection conn = (HttpURLConnection) new URL(this.instance + params)
                 .openConnection();
         conn.setInstanceFollowRedirects(false);
-        conn.setRequestProperty("Cookie", "dosid=" + sid);
-        return conn.getResponseCode();
+        conn.setRequestProperty("Cookie", "dosid=" + this.sid);
+        return conn;
+    }
+
+    public String getDataInventory(String params){
+        String data = null;
+        try {
+            HttpURLConnection conn = getConnection(params);
+            conn.setRequestProperty("Content-Type","application/x-www-form-urlencoded");
+            BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
```

Comment:
Reading the stream into the base64 decode should be part of the util. All of this code should be:

`data = Base64Utils.base64Decode(conn.getInputStream());`
Let Base64Utils have a method that reads an InputStream, converts to string, and passes it over to the "string to string" base64 decode method